File: SyntaxPractice/problem-1/service/Service.py
from model.Result import Result
from exception.InputValidationException import InputValidationException


class Service:

    def process(self, numbers):

        if numbers is None or len(numbers) == 0:
            raise InputValidationException(
                "Input list cannot be empty"
            )

        total = sum(numbers)

        average = (
            total / len(numbers)
        )

        return Result(
            total,
            average
        )

File: SyntaxPractice/problem0/service/Service.java
package service;

import exception.InputValidationException;
import model.Result;

import java.util.List;

public class Service {

    public Result process(List<Integer> numbers) {

        if (numbers == null || numbers.isEmpty()) {
            throw new InputValidationException("Input list cannot be empty");
        }

        int sum = numbers.stream()
                .mapToInt(Integer::intValue)
                .sum();

        double average = (double) sum / numbers.size();

        return new Result(sum, average);
    }
}
